spec/services/online_codes_import_service_spec.rb

```diff
@@ -0,0 +1,33 @@
+require 'rails_helper'
+
+RSpec.describe OnlineCodesImportService do
+  context 'when importing online codes from csv file' do
+    it 'imports all online codes from csv file' do
+      create(:reward, title: 'RewardTest1')
+      create(:reward, title: 'RewardTest2')
+      params = { file: fixture_file_upload('./spec/fixtures/online_codes/valid_online_codes_import.csv') }
+      import_online_codes = described_class.new(params)
+
+      expect(OnlineCode.count).to eq 0
+      expect(import_online_codes.import).to be true
+      expect(OnlineCode.count).to eq 2
```

Comment:
it won't always be true. There are some gimmics when it comes to running tests in real pipelines - database is not always cleared down to no resources before each test.

Also, testing for count being 0 is pointless from the perspective of the production code, as it tests the test itself

The test should look more or less like this (written from head, forgive typos please)
```ruby

# initiate the variable here so that it is overwritten in the block below, 
# otherwise `result = ...` in the block wouldn't be visible outside of the block
result = nil 

# pass block to `expect` to be able to measure changes within execution
expect { result = import_online_codes.import }.to change(OnlineCode, :count).by(2)
expect(result).to be true
```

Reply:
Got it 👍  I have changed it according to your suggestion :) Thanks!